### PR TITLE
Add NetworkServiceServer chain element for handling Mechanism Selection

### PR DESCRIPTION
**Pull request by edwarnicke**

### CODE REVIEW

File: pkg/networkservice/common/mechanisms/server.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+func (m mechanismsServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
+	if request.GetConnection().GetMechanism() == nil {
+		for _, mechanism := range request.GetMechanismPreferences() {
+			srv, ok := m.mechanisms[mechanism.GetType()]
+			if ok {
+				req := request.Clone()
+				req.GetConnection().Mechanism = mechanism
+				conn, err := srv.Request(ctx, req)
+				if err == nil {
+					request = request.Clone()
```
Comment (fkautz) [inline pkg/networkservice/common/mechanisms/server.go]: What's the reason for the second clone here?

File: pkg/networkservice/common/mechanisms/server.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+					request.Connection = conn
+					return next.Server(ctx).Request(ctx, request)
```
Comment (fkautz) [inline pkg/networkservice/common/mechanisms/server.go]: Did you mean to return here? Is L61 only meant to run if no preferences are found?

File: pkg/networkservice/common/mechanisms/server.go (hunk 1 above)
Comment (edwarnicke) [reply pkg/networkservice/common/mechanisms/server.go]: You are correct.

File: pkg/networkservice/common/mechanisms/server.go (hunk 2 above)
Comment (edwarnicke) [reply pkg/networkservice/common/mechanisms/server.go]: Yes, but your question led me to rewriting in a way that is hopefully clearer.

**Pull request merged**